src/txkube/testing/strategies.py

```diff
@@ -299,49 +299,47 @@ def deployments():
     )
 
 
-def deploymentlists():
+def _collections(cls, strategy, unique_by):
     """
-    Strategy to build ``DeploymentList``.
+    A helper for defining a strategy to build ``...List`` objects.
+
+    :param cls: The model class to instantiate.
+    :param strategy: A strategy to build elements for the collection.
+    :param unique_by: A key function compatible with the ``lists`` strategy.
     """
     return builds(
-        v1beta1.DeploymentList,
-        items=lists(
-            deployments(),
-            average_size=_QUICK_AVERAGE_SIZE,
-            max_size=_QUICK_MAX_SIZE,
-            unique_by=_unique_names_with_namespaces,
+        cls,
+        items=one_of(
+            none(),
+            lists(
+                strategy,
+                average_size=_QUICK_AVERAGE_SIZE,
+                max_size=_QUICK_MAX_SIZE,
+                unique_by=unique_by,
+            ),
         ),
     )
 
 
+def deploymentlists():
+    """
+    Strategy to build ``DeploymentList``.
+    """
+    return _collections(v1beta1.DeploymentList, deployments(), _unique_names_with_namespaces)
```

Comment:
in line 311 the code is generalized to not have `v1beta1`, whereas this is still using a fixed/hardcoded version; worth generalizing into a specific test avenue?